### PR TITLE
Receive ecc device keys from Console

### PR DESCRIPTION
**Pull request by hwahby**
Will move to Console generating keys needed to interact with AWS, instead of generating on Router.

https://github.com/helium/console/issues/1088
https://github.com/helium/router/issues/745

### CODE REVIEW

File: src/device/router_device_worker.erl [hunk 1]
```diff
@@ -374,18 +374,28 @@ handle_cast(
...
-                        lorawan_rxdelay:maybe_update(APIDevice, Device0),
-                        router_device:metadata(APIDevice)
-                    )},
-                {is_active, IsActive}
-            ],
+            ECCUpdate =
+                case router_device:ecc_compact(APIDevice) of
+                    undefined ->
+                        [];
+                    ECC ->
```
Comment (michaeldjeffrey) [inline src/device/router_device_worker.erl:381]: From the device record.
`ecc_compact :: map() | undefined,`

I think that's going to break `base64:decode/1`

File: src/device/router_device_worker.erl [hunk 2]
```diff
@@ -374,18 +374,28 @@ handle_cast(
...
+                    {dev_eui, router_device:dev_eui(APIDevice)},
+                    {app_eui, router_device:app_eui(APIDevice)},
+                    {devaddrs, DevAddrs},
+                    {metadata,
+                        maps:merge(
+                            lorawan_rxdelay:maybe_update(APIDevice, Device0),
+                            router_device:metadata(APIDevice)
+                        )},
+                    {is_active, IsActive}
+                ] ++ ECCUpdate,
```
Comment (michaeldjeffrey) [inline src/device/router_device_worker.erl:398]: We might need to move this to the api worker so it comes in with every API device.

**Comment by macpie** [comment]
We will be doing the migration in here as well. 

`router_device_worker:get_device(DB, CF, DeviceID)` (the one used in `init/1`) should check for keys from `Device1` and compare with keys from disk (`Device0`) if they are different Router should ignore Console and also make an API call to Console to update the keys

File: src/device/router_device_worker.erl [hunk 3]
```diff
@@ -374,18 +374,28 @@ handle_cast(
...
-                    )},
-                {is_active, IsActive}
-            ],
+            ECCUpdate =
+                case router_device:ecc_compact(APIDevice) of
+                    undefined ->
+                        [];
+                    ECC ->
+                        DecodeECC = base64:decode(ECC),
+                        [{ecc_compact, libp2p_crypto:bin_to_pubkey(DecodeECC)}]
```
Comment (macpie) [inline src/device/router_device_worker.erl:383]: here you will need to use `libp2p_crypto:keys_from_bin/1`

**Pull request closed without merging**